### ISSUE TITLE
Class constructor Model cannot be invoked without 'new'

### ISSUE THREAD

**Issue by rolandleth** [sequelize/sequelize#11270]
Hello!

## What are you doing?
```js
const sequelize = new Sequelize(
  process.env.DATABASE_URL,
  {
    dialect: "postgres",
    logging: false,
  }
)

class Account extends Model {
  id!: number
  name!: string
}

Account.init(
  {
      name: {
        type: new DataTypes.STRING(128),
        allowNull: false,
      }
  },
  {
      tableName: "Account",
      sequelize,
    }
)

const acc = await Account.create({
    name: "Roland Leth",
})
```

**To Reproduce**
1. Define model Account with `class` approach. `sequelize.define` works properly.
2. Try and create/update an entry. Read/delete queries work.
3. See the error and stack trace attached.

## What do you expect to happen?
The record to be created/updated.

## What is actually happening?
It throws `Class constructor Model cannot be invoked without 'new'`.

## Environment
Dialect:
- [ ] mysql
- [ ] postgres
- [ ] sqlite
- [ ] mssql
- [x] any
Dialect **library** version: 7.12
Database version: 11
Sequelize version: 5.12.2
Node Version: 12.6
OS: 10.15
If TypeScript related: TypeScript version: 3.6.0-dev.20190801 (3.5.x behaves the same)
Tested with latest release:
- [ ] No
- [x] Yes, specify that version: 5.12.2

I've read all that I could find about this error, including other issues here, but my lack of web experience might be at fault.

I'm using [razzle](https://github.com/jaredpalmer/razzle), if that helps in any way.

This is my `.babelrc`:
```js
{
  "presets": [
    "razzle/babel", // NEEDED
    [
      "@babel/preset-env",
      {
        // "spec": true,
        // "debug": true,
        "useBuiltIns": "usage",
        "corejs": "3.1.4"
      }
    ]
  ]
}

```
And [this](https://github.com/jaredpalmer/razzle/blob/fad9cc53a3d433ea299b26c7ece08131053e6d1e/packages/babel-preset-razzle/index.js) is razzle's.

## Stack trace
```bash
path-to-project/build/server.js:6985
    return _babel_runtime_helpers_possibleConstructorReturn__WEBPACK_IMPORTED_MODULE_1___default()(this, _babel_runtime_helpers_getPrototypeOf__WEBPACK_IMPORTED_MODULE_2___default()(Account).apply(this, arguments));
                                                                                                                                                                                               ^

TypeError: Class constructor Model cannot be invoked without 'new'

  6983 |     _babel_runtime_helpers_classCallCheck__WEBPACK_IMPORTED_MODULE_0___default()(this, Account);
  6984 | 
> 6985 |     return _babel_runtime_helpers_possibleConstructorReturn__WEBPACK_IMPORTED_MODULE_1___default()(this, _babel_runtime_helpers_getPrototypeOf__WEBPACK_IMPORTED_MODULE_2___default()(Account).apply(this, arguments));
  6986 |   }
  6987 | 
  6988 |   return Account;

    at new Account (path-to-project/build/server.js:6985:192)
    at Function.build (path-to-project/node_modules/sequelize/lib/model.js:2164:12)
    at Function.create (path-to-project/node_modules/sequelize/lib/model.js:2217:17)
    at Module../src/server.ts (path-to-project/src/server.ts:7:1)
    at __webpack_require__ (path-to-project/build/webpack:/webpack/bootstrap:685:1)
    at fn (path-to-project/build/webpack:/webpack/bootstrap:59:1)
    at Module../src/index.ts (path-to-project/build/webpack:/src/index.ts:1:1)
    at __webpack_require__ (path-to-project/build/webpack:/webpack/bootstrap:685:1)
    at fn (path-to-project/build/webpack:/webpack/bootstrap:59:1)
    at Object.0 (path-to-project/build/server.js:8034:18)
    at __webpack_require__ (path-to-project/build/webpack:/webpack/bootstrap:685:1)
    at path-to-project/build/webpack:/webpack/bootstrap:752:1
    at Object.<anonymous> (path-to-project/build/server.js:757:10)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Function.Module.runMain (internal/modules/cjs/loader.js:839:10)
    at internal/main/run_main_module.js:17:11
```

Please let me know what other info I can provide.

Thank you!

**Comment by papb** [comment]
Hello! Your code snippet does not seem self-contained - what is `db`? It is not defined anywhere I can see...

**Comment by rolandleth** [comment]
My bad, just an artifact remaining after moving some code to provide the snippet. Should be just `Account`. Updated.

**Comment by papb** [comment]
I was able to reproduce. Micro SSCCE/MCVE/reprex:

```ts
// test.ts
const Sequelize = require("sequelize");
const sequelize = new Sequelize('sqlite::memory:');

class AccountTest extends Sequelize.Model {
    id!: number
    name!: string
}
AccountTest.init({ name: Sequelize.STRING(128) }, { sequelize });

(async () => {
    await sequelize.sync();
    await AccountTest.create({ name: "Foo" });
})();
```

Executing `ts-node test.ts`:

`TypeError: Class constructor Model cannot be invoked without 'new'`

**Comment by papb** [comment]
@SimonSchick Can you take a look later?

**Comment by SimonSchick** [comment]
Duplicate of https://github.com/sequelize/sequelize/issues/10783

**Comment by rolandleth** [comment]
Hey, @SimonSchick, what should I do now? It's a duplicate, but of an issue that is closed and whose solution didn't help me — I already tried it before opening this one 😞

**Comment by SimonSchick** [comment]
Again I'm not exactly an expert when it comes to babel, but can you tell me why you need babel on node 12.6?
Since it seems you want to use types I'd strongly recommend using the typescript compiler instead.

**Comment by papb** [comment]
@SimonSchick Babel is not involved, see the micro SSCCE I created. I did not use babel (unless ts-node uses babel under the hood and I don't know)

**Comment by SimonSchick** [comment]
@papb please post your stacktrace.

**Comment by SimonSchick** [comment]
`ts-node` uses `es5` as a transpilation target, which is probably the issue here, you need to specify at least `es6`, but again this isn't a sequelize issues, you need to create an environment which is at least equivalent to node v6+ for sequelize v5.

**Comment by rolandleth** [comment]
@SimonSchick The project also contains a React app and ... with my limited web knowledge, that's what I managed to come up with (regarding using babel). Regarding targets, I do have `target: "esnext"` in my `tsconfig.json` (tried with `es6` as well, same results), does that matter?

**Comment by papb** [comment]
@SimonSchick Interesting. I didn't know that. Indeed, adding the following `tsconfig.json` to my root folder fixed it:

```js
// tsconfig.json
{
    "compilerOptions": {
        "target": "es6"
    }
}
```

Output:

```
Executing (default): CREATE TABLE IF NOT EXISTS `AccountTests` (`id` INTEGER PRIMARY KEY AUTOINCREMENT, `name` VARCHAR(128), `createdAt` DATETIME NOT NULL, `updatedAt` DATETIME NOT NULL);
Executing (default): PRAGMA INDEX_LIST(`AccountTests`)
Executing (default): INSERT INTO `AccountTests` (`id`,`name`,`createdAt`,`updatedAt`) VALUES (NULL,$1,$2,$3);
```

Thanks :+1:

**Comment by papb** [comment]
> @SimonSchick The project also contains a React app and ... with my limited web knowledge, that's what I managed to come up with (regarding using babel). Regarding targets, I do have `target: "esnext"` in my `tsconfig.json` (tried with `es6` as well, same results), does that matter?

As my experiments above show, indeed it matters, but I was able to fix my micro SSCCE with `"target": "es6"`. If your problems still persist after that, then the fact that you're using other stuff like babel/react/razzle might be the cause. Unfortunately I know very little about those, so I won't be able to help further.

Perhaps you should try asking a question in StackOverflow? The `sequelize.js` tag is available.

**Comment by rolandleth** [comment]
Yupp, will do that since it still persists and it's clearly out of scope. 

Thank you both for your help and input, have a great weekend!

**Comment by papb** [comment]
@rolandleth When you do, add a link on your question to this issue, and also post a link of the question here, so that other in a similar situation can find help faster. Thanks!

**Comment by rolandleth** [comment]
Both luckily (because I solved it) and sadly (because I don't have a solution for _this_ problem), I used @SimonSchick's suggestion to stop using Babel and moved to TS' compiler. Everything is working properly now. 

If I'll ever need Babel and have to fix this, I'll be sure to come back and update this as well.

**Comment by 103cuong** [comment]
Hi guys, I have the same issue with tsc (TypeScript compiler).

### version

```json
{
  "sequelize": "^5.21.6",
  "ts-node": "^8.10.1",
  "typescript": "^3.7.2"
}
```

### model

```typescript
import { Model, DataTypes } from 'sequelize';
import sequelize from './';
import { OrderStatus } from '../components/constants';

class Order extends Model {
  public id!: number;
  public total!: number;
  public finalTotal!: number;
  public status: OrderStatus;
  public start_date: Date;
  public end_date: Date;
  public created_at: Date;
  public updated_at: Date;
}

Order.init({
  id: {
    type: DataTypes.UUID,
    primaryKey: true,
    defaultValue: DataTypes.UUIDV4
  },
  total: {
    type: DataTypes.INTEGER,
    allowNull: true,
  },
  finalTotal: {
    type: DataTypes.INTEGER,
    allowNull: true,
    validate: {
      len: {
        args: [2, 5],
        msg: 'Name must be from 2 to 5 characters in length'
      },
    }
  },
  status: {
    type: DataTypes.ENUM(OrderStatus.InProcess, OrderStatus.Done)
  },
}, {
  sequelize,
  tableName: 'orders',
})

export default Order;
```

### sequelize.import

```typescript
const modelsPath = `${__dirname}/../src/models`;
  const models = fs
    .readdirSync(modelsPath)
    .filter((filename: string) => /model.ts$/.test(filename))
    .reduce((total: any, filename: string) => {
      const model = sequelize.import(path.join(modelsPath, filename));
      total[capitalize(model.name)] = model;
      return total;
    }, {});

  // Sets up the associations for each model.
  Object.keys(models).forEach((modelName: string) => {
    if ('associate' in models[modelName]) {
      models[modelName].associate(models);
    }
  });
```

### error

```sh
TypeError: Class constructor Order cannot be invoked without 'new'
      at Sequelize.import (node_modules/sequelize/lib/sequelize.js:486:38)
      at /home/cuongw/Workspace/SaveMoney/service-template/test/utils.ts:18:37
      at Array.reduce (<anonymous>)
      at Object.exports.prepareModels (test/utils.ts:17:6)
      at Object.exports.loadFixtures (test/utils.ts:35:18)
      at Context.<anonymous> (test/order.test.ts:16:11)
      at processImmediate (internal/timers.js:456:21)
```

Please tell me a solution. Thank all.

**Comment by divye11** [comment]
@harrytran103 Did you find a solution for this?

**Comment by dimadolgopolovn** [comment]
For me, it was a **ts-node issue** (still unresolved). When I compile everything with tsc for testing, it works perfectly.

The funny thing is that it appeared only after I added the 5th model. It uses the same code structure to introduce it, and compiles it the same way, but I start getting the same error.

Below, anything before Message including it works. Preregistration number doesn't.

`
...
class Message extends Model {
  ...
}

Message.init(
  {
    ...
  },
  { sequelize, modelName: "message" },
)

class PreregistrationNumber extends Model {
  ...
}

PreregistrationNumber.init(
  ...
  { sequelize, modelName: "preregistrationNumber" },
)`
